lib/omnibus-ctl.rb

```diff
@@ -506,6 +512,66 @@ def reconfigure(exit_on_success=true)
       end
     end
 
+    def check_license_acceptance(override_accept = false)
+      license_guard_file_path = File.join(data_path, ".license.accepted")
+
+      # If the project does not have a license we do not have
+      # any license to accept.
+      return unless File.exist?(project_license_path)
+
+      if !File.exist?(license_guard_file_path)
+        if override_accept || ask_license_acceptance
+          FileUtils.touch(license_guard_file_path)
+        else
+          log "Please accept software's license in order to continue."
+          exit(1)
+        end
+      end
+    end
+
+    def ask_license_acceptance
+      log "To use this software, you must agree to the terms of the software license agreement."
+
+      if !STDIN.tty?
+        log "You need to either view and agree to the license agreement or pass --accept-license."
+        exit(1)
+      end
+
+      loop do
+        log "Press 'Return' to view the software license agreement, 'q' to quit."
+
+        user_input = STDIN.getch
+        user_input << STDIN.getch while STDIN.ready?
+        case user_input
+        when "\n", "\r", "\r\n"
+          break
+        when "q"
+          return false
+        else
+          log "Sorry, I do not understand '#{user_input}'."
+        end
+      end
+
+      system("less #{project_license_path}")
+
+      loop do
+        log "Type 'yes' to agree to the license agreement, or anything else to cancel."
+
+        user_input = gets.chomp.downcase
+        case user_input
+        when "yes"
+          return true
+        else
+          log "You have not agreed to the license agreement."
```

Comment:
To @oferrigni's point, if someone mis-types yes and hits return, we'll not accept the license, but it's not necessarily clear how to then accept the license. So the suggestion is to add a line so it reads something like this:

You have not agreed to the license agreement.
If you would like to agree, please re-run reconfigure to trigger this prompt again.

Something like that.